### PR TITLE
Decimals widgets

### PR DESCRIPTION
**Pull request by thib66**
changes for more consistency with other widgets like dial widget

### CODE REVIEW

**Comment by Paul-Reed** [comment]
Aymeric, Out of interest. was this in response  to changing the comparison operators in the 'feedvalue' widget  from == to === (that it did not work because it was comparing a number and a string - not equal types).

Paul

File: widget/dewpoint/dewpoint_render.js [hunk 1]
```diff
@@ -211,6 +211,7 @@ function draw_dewpoint(context,
 					else {
 						val = val.toFixed(2);
 						}
+				    val = parseFloat(val);
```
Comment (Paul-Reed) [inline widget/dewpoint/dewpoint_render.js]: Mixture of tabs & spaces in indent

File: widget/frostpoint/frostpoint_render.js [hunk 2]
```diff
@@ -206,6 +206,7 @@ function draw_frostpoint(context,
 					else {
 						val = val.toFixed(2);
 						}
+				 val = parseFloat(val);
```
Comment (Paul-Reed) [inline widget/frostpoint/frostpoint_render.js]: Mixture of tabs & spaces in indent

**Comment by thib66** [comment]
Hello, I got some issues with decimals but I am unable to reproduce the problem.
My apologies,
Aymeric

**Comment by Paul-Reed** [comment]
No apologies needed, I think that these changes should be made to ensure that they are equal types (numbers and not strings). Do you agree @TrystanLea or @glynhudson ?

**Comment by Paul-Reed** [comment]
Aymeric, I've been looking back through where I had experienced this issue previously, and cannot re-create now. Shall we close this now as there has been no further comments made, and we can always revisit this later if we get the same problems again.

Paul

**Comment by thib66** [comment]
Hello Paul,
Yes you can close the PR.
Thanks,
Aymeric

**Comment by thib66** [comment]
Hello 
@Paul-Reed, I know now why parseFloat is used.
Of course the val variable is in numeric format so it seems not necessary to use parseFloat(). 
But look at the behaviour of the parsefloat function below : 
if val = 9.51 then parseFloat(val) returns 9.51 (no mystery)
if now val = 9.50 then parseFloat(val) returns 9.5
In fact parseFloat() function is used to delete non significant digits (in other terms last zeros in decimals are suppressed)
The parseFloat() is actually used with dial widget for the decimals option set to "Automatic".
I think that whe have to apply this function for all the widgets where we have the choice to select the number of decimals.
Regards,
Aymeric

**Comment by Paul-Reed** [comment]
Aymeric, I'm out of the country, with no system access, so hopefully one of the other emoncms team members will pick this up.
Otherwise, I'm back on the 29th June.

Regards & thanks for your support.

Paul

**Comment by Paul-Reed** [comment]
Yes, agree Aymeric. It makes sense to have consistency across the widgets, and remove the non-significant digits when decimals are set as automatic. Good spot.

**Pull request closed without merging**